src/mol-geo/geometry/color-data.ts

```diff
@@ -156,11 +157,19 @@ function createInstanceColor(locationIt: LocationIterator, color: LocationColor,
 /** Creates color texture with color for each group (i.e. shared across instances) */
 function createGroupColor(locationIt: LocationIterator, color: LocationColor, colorData?: ColorData): ColorData {
     const { groupCount } = locationIt;
-    const colors = createTextureImage(Math.max(1, groupCount), 3, Uint8Array, colorData && colorData.tColor.ref.value.array);
+    const colors = createTextureImage(Math.max(1, groupCount * 2), 3, Uint8Array, colorData && colorData.tColor.ref.value.array);
```

Comment:
We need to find a way to only double the size when using. Also I don't like haveing a specific type of location exposed here. One idea is to add an optional getLocation2 to the LocationIterator and a property hasLocation2 which you can use here.

Reply:
@arose I'm trying to implement this logic, I need a way to update ```createLocationIterator``` or pass an additional variable to it in order to use single or double location iterator depending on the ```colorMode``` property. Do you think it is possible in some way?

Reply:
@arose, this should be done with last commit.

Reply:
thanks, looks good, just did a few tweaks to the locate-iterator to consolidate the code